src/helpers/helpers.dom.ts

```diff
@@ -65,27 +69,34 @@ function getPositionedStyle(styles, style, suffix) {
   return result;
 }
 
-const useOffsetPos = (x, y, target) => (x > 0 || y > 0) && (!target || !target.shadowRoot);
+const useOffsetPos = (x: number, y: number, target: HTMLElement | EventTarget) =>
+  (x > 0 || y > 0) && (!target || !(target as HTMLElement).shadowRoot);
 
 /**
  * @param {Event} e
  * @param {HTMLCanvasElement} canvas
  * @returns {{x: number, y: number, box: boolean}}
  */
-function getCanvasPosition(e, canvas) {
-  // @ts-ignore
-  const touches = e.touches;
+function getCanvasPosition(
+  e: Event | TouchEvent | MouseEvent,
+  canvas: HTMLCanvasElement
+): {
+    x: number;
+    y: number;
+    box: boolean;
+  } {
+  const touches = (e as TouchEvent).touches;
   const source = touches && touches.length ? touches[0] : e;
```

Comment:
maybe better to cast type once
```suggestion
  const source = (touches && touches.length ? touches[0] : e) as MouseEvent;
```